docs/namespaces.md

```diff
@@ -0,0 +1,126 @@
+---
+title: "Paths and namespaces"
+date: 2020-08-27T11:08:00+01:00
+weight: 42
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs
+geekdocFilePath: namespaces.md
+---
+
+{{< toc >}}
+
+When a client makes a request to OCIS the path encoded in the request needs to undergo several transformations before actually hitting the storage.
+Discussing the related code and request flow is unnecessarily hard without a clear terminology.
+Clarifying the different endpoints, involved services and path transformations creates a common vocabulary. This page serves as an  example and forms the glossary for paths, namespaces and related concepts.
+
+## Endpoints
+
+OCIS implements the two webdav endpoints that are present in ownCloud 10:
+- `/dav/files/<userid>/` which presents the tree of the user with the given user id.
+- `/webdav/` which presents the tree of the currently logged in user. It [does not allow referencing another users files](https://github.com/owncloud/core/issues/12504#issuecomment-65259957) and only exists for [backwards compatability](https://github.com/owncloud/core/issues/12543).
+
+For backwards compatapility you can add the old `remote.php`, so these also work:
+- `/remote.php/dav/files/<userid>/`
+- `/remote.php/webdav/`
+
+## Paths
+
+Whatever follows the two endpoints is the *relative path* of a file in a users home storage. These two URLs actually point to the same file for the user with the id `u-u-i-d`:
+- `https://cloud.example.org/dav/files/<userid>/path/to/file.md`
+- `https://cloud.example.org/webdav/path/to/file.md`
+
+Since `path/to/file.md` is only relative to the users home storage we need to prefix it with a path to the users home storage in the CS3 namespace.
+
+## Services
+
+For WebDAV requests:
+
+`client -webdav-> proxy -webdav-> ocdav -cs3-> reva gateway -cs3-> reva storage provider`
```

Comment:
is `-webdav` meant to be `->webdav` ? the dash here looks confusing.
maybe put these in parenthesis?

Reply:
ah, got it now, it's part of the arrow. then maybe make the arrow longer: `--webdav-->` or even `--(webdav)-->`

Reply:
creating a diagram for this